backends/memcheck/ceed-memcheck-vector.c

```diff
@@ -49,15 +49,21 @@ static int CeedVectorSetArray_Memcheck(CeedVector vec, CeedMemType mem_type, Cee
 
   CeedCheck(mem_type == CEED_MEM_HOST, CeedVectorReturnCeed(vec), CEED_ERROR_BACKEND, "Can only set HOST memory for this backend");
 
+  if (impl->array_allocated) {
+    for (CeedSize i = 0; i < length; i++) impl->array_allocated[i] = NAN;
+  }
```

Comment:
How setting array to `NAN` helps?

Reply:
```C
int main(int argc, char **argv) {
  Ceed       ceed;
  CeedVector x;
  CeedInt    len = 10;

  CeedInit(argv[1], &ceed);

  CeedVectorCreate(ceed, len, &x);
  CeedVectorSetValue(x, 1.0);

  {
    const CeedScalar *read_array, *misbehavior;

    CeedVectorGetArrayRead(x, CEED_MEM_HOST, &read_array);
    misbehavior = read_array;
    CeedVectorRestoreArrayRead(x, &read_array);

    // I want this to show garbage because we've violated access assumptions
    for (CeedInt i = 0; i < len; i++) {
      printf("%f\n", misbehavior[0]);
    }
  }

  CeedVectorDestroy(&x);
  CeedDestroy(&ceed);
  return 0;
}
```